src/main/resources/frontend/src/services/submissionService.ts

```diff
@@ -1,105 +1,41 @@
 import type {Submission} from "@/types/types";
 import { Phase } from "@/types/types";
-import {useAppConfigStore} from "@/stores/appConfig";
+import { ServerCommunicator } from '@/network/ServerCommunicator'
 
 export const submissionsGet = async (phase: Phase | null): Promise<Submission[]> => {
-    let url = useAppConfigStore().backendUrl + '/api/submission' + (phase === null ? "" : "/" + Phase[phase])
-    const response = await fetch(url, {
-        method: 'GET',
-        credentials: 'include'
-    });
-
-    if (!response.ok) {
-        console.error(response);
+    try{
+        return await ServerCommunicator.getRequest<Submission[]>('/api/submission' + (phase === null ? "" : "/" + Phase[phase]))
```

Comment:
See all of these methods doing the same `try ... catch ... return default value` pattern? As previously mentioned, a function would really do some good here.

Reply:
A of the methods I replaced had the try, catch, console.error() the error. `ServerCommunicator` will now log the error automatically (but also throw something). I played around with making a `default value` parameter to the request methods, but I couldn't decide if that made the API too complicated. In some cases, it could look like
`ServerCommunicator.postRequest("/api/endpoint", { body }, false, { default empty})`. Would we want to make that default value a part of the request methods as an optional parameter? or a separate function?